### ISSUE TITLE
Attempt to access disallowed key feature_* in a restricted hash

### ISSUE THREAD

**Issue by atoomic** [rurban/perl-compiler#250]
with perl 5.22
this is an invalid state usage, but this should die and not abort

````
> perl -e 'use feature ":5.10"; eval q/state ($a) = 1; 1/ or print qq/ok\n/;'
ok
> perlcc -r -e 'use feature ":5.10"; eval q/state ($a) = 1; 1/ or print qq/ok\n/;'
pccXoM7K.c:9746: warning: this decimal constant is unsigned only in ISO C90
pccXoM7K.c:9746: warning: this decimal constant is unsigned only in ISO C90
pccXoM7K.c:9747: warning: this decimal constant is unsigned only in ISO C90
pccXoM7K.c:9747: warning: this decimal constant is unsigned only in ISO C90
pccXoM7K.c:9748: warning: this decimal constant is unsigned only in ISO C90
pccXoM7K.c:9748: warning: this decimal constant is unsigned only in ISO C90
pccXoM7K.c:10445: warning: integer constant is too large for ‘unsigned long’ type
pccXoM7K.c:10446: warning: integer constant is too large for ‘unsigned long’ type
Attempt to access disallowed key 'feature_switch' in a restricted hash.
````

**Comment by atoomic** [comment]
similar issue with this
````
> perl -e 'my $function = "uc"; my $char = "\xff"; use feature "unicode_strings"; print qq/ok\n/ if ord uc($char) eq ord eval "$function(q/$char/)", "\n";'
ok
# the error can randomly be one or the other
> perlcc -r -e 'my $function = "uc"; my $char = "\xff"; use feature "unicode_strings"; print qq/ok\n/ if ord uc($char) eq ord eval "$function(q/$char/)", "\n";'
Attempt to access disallowed key 'feature_unicode' in a restricted hash.
> perlcc -r -e 'my $function = "uc"; my $char = "\xff"; use feature "unicode_strings"; print qq/ok\n/ if ord uc($char) eq ord eval "$function(q/$char/)", "\n";'
Attempt to access disallowed key 'feature_unicode' in a restricted hash.
````

**Comment by atoomic** [comment]
another instance of this problem
````
> perl -e 'use feature q/evalbytes/; print "ok\n" if evalbytes("1+7") == 8'
ok
> perlcc -r -e 'use feature q/evalbytes/; print "ok\n" if evalbytes("1+7") == 8'
Attempt to access disallowed key 'feature_evalbytes' in a restricted hash.
````

**Comment by atoomic** [comment]
````
> perl -e 'use feature q.refaliasing.; no warnings; eval q{\$x = \$y}; print qq/ok\n/'
ok
> perlcc -e 'use feature q.refaliasing.; no warnings; eval q{\$x = \$y}; print qq/ok\n/'; ./a.out
Attempt to access disallowed key 'feature_refaliasing' in a restricted hash.
````

**Comment by atoomic** [comment]
````
> perlcc -r -e 'use utf8; use open qw( :utf8 :std ); print qq/ok\n/'
pcc_cUHF.c:10724: warning: this decimal constant is unsigned only in ISO C90
pcc_cUHF.c:10724: warning: this decimal constant is unsigned only in ISO C90
pcc_cUHF.c:10725: warning: this decimal constant is unsigned only in ISO C90
pcc_cUHF.c:10725: warning: this decimal constant is unsigned only in ISO C90
pcc_cUHF.c:10726: warning: this decimal constant is unsigned only in ISO C90
pcc_cUHF.c:10726: warning: this decimal constant is unsigned only in ISO C90
pcc_cUHF.c:11340: warning: integer constant is too large for ‘unsigned long’ type
pcc_cUHF.c:11341: warning: integer constant is too large for ‘unsigned long’ type
ok
> perlcc -r -e 'use utf8; { package ꕽ::Ʉ; 1; } eval "sub { my ꕽ \$obj = shift; }"; print qq/ok\n/'
ok
> perlcc -r -e 'use utf8; use open qw( :utf8 :std ); { package ꕽ::Ʉ; 1; } eval "sub { my ꕽ \$obj = shift; }"; print qq/ok\n/'
pccBiK9Y.c:10741: warning: this decimal constant is unsigned only in ISO C90
pccBiK9Y.c:10741: warning: this decimal constant is unsigned only in ISO C90
pccBiK9Y.c:10742: warning: this decimal constant is unsigned only in ISO C90
pccBiK9Y.c:10742: warning: this decimal constant is unsigned only in ISO C90
pccBiK9Y.c:10743: warning: this decimal constant is unsigned only in ISO C90
pccBiK9Y.c:10743: warning: this decimal constant is unsigned only in ISO C90
pccBiK9Y.c:11357: warning: integer constant is too large for ‘unsigned long’ type
pccBiK9Y.c:11358: warning: integer constant is too large for ‘unsigned long’ type
Attempt to access disallowed key 'open<' in a restricted hash.
````

**Comment by atoomic** [comment]
if we force the hash to be set during compilation time, then it can be compiled
````
> perlcc -r -e 'BEGIN { use feature ":5.10"; } eval q/state ($a) = 1; 1/ or print qq/ok\n/;'
ok
````

**Comment by rurban** [comment]
So the hash key is not found in the feature hash, which is a big issue. I think we have to use  hv_storehek with the hek instead of hv_store of the string.

**Comment by rurban** [comment]
Fixed with B::C 1.52_08, merged branch feature-gh250
